### PR TITLE
Escape underscores in paradox unidoc directives

### PR DESCRIPTION
**Pull request by raboof**
Refs #1648

### CODE REVIEW

File: docs/src/main/paradox/scala/http/common/http-model.md [hunk 1]
```diff
@@ -134,7 +134,7 @@ The stream consists of zero or more @scala[`Chunked`]@java[non-empty chunks] par
 
 @scala[HttpEntity.CloseDelimited]@java[HttpEntityCloseDelimited]
 : An unchunked entity of unknown length that is implicitly delimited by closing the connection (`Connection: close`).
-The content data are presented as a @scala[`Source[ByteString]`]@java[@unidoc[Source[ByteString, ?]]].
+The content data are presented as a @scala[@unidoc[Source[ByteString]]]@java[@unidoc[Source[ByteString, ?]]].
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/common/http-model.md]: `@unidoc[Source[ByteString, NotUsed]]`?

File: docs/src/main/paradox/scala/http/common/http-model.md [hunk 2]
```diff
@@ -159,7 +159,7 @@ The @scala[@unidoc[HttpEntity] companion object]@java[class `HttpEntities`] cont
 You can @scala[pattern match over]@java[use] the @scala[subtypes]@java[`isX` methods] of @unidoc[HttpEntity] @java[to find out of which subclass an entity is] if you want to provide
 special handling for each of the subtypes. However, in many cases a recipient of an @unidoc[HttpEntity] doesn't care about
 of which subtype an entity is (and how data is transported exactly on the HTTP layer). Therefore, the general method
-@scala[`HttpEntity.dataBytes`]@java[`HttpEntity.getDataBytes()`] is provided which returns a @scala[`Source[ByteString, Any]]]@java[@unidoc[Source[ByteString, ?]]] that allows access to the data of an
+@scala[`HttpEntity.dataBytes`]@java[`HttpEntity.getDataBytes()`] is provided which returns a @scala[@unidoc[Source[ByteString, Any]]]@java[@unidoc[Source[ByteString, ?]]] that allows access to the data of an
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/common/http-model.md]: `@unidoc[Source[ByteString, NotUsed]]`?

File: docs/src/main/paradox/scala/http/common/http-model.md [hunk 3]
```diff
@@ -121,7 +121,7 @@ It wraps a plain @unidoc[ByteString] and  represents a standard, unchunked entit
 
 @scala[HttpEntity.Default]@java[HttpEntityDefault]
 : The general, unchunked HTTP/1.1 message entity.
-It has a known length and presents its data as a @scala[@unidoc[Source[ByteString]`]@java[@unidoc[Source[ByteString, ?]]] which can be only materialized once.
+It has a known length and presents its data as a @scala[@unidoc[Source[ByteString]]]@java[@unidoc[Source[ByteString, ?]]] which can be only materialized once.
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/common/http-model.md]: `@unidoc[Source[ByteString, NotUsed]]`?

File: docs/src/main/paradox/scala/http/migration-guide/migration-from-spray.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ Akka HTTP offers an API based on streams where spray offered an API based on act
 Streaming support is needed to handle request and response entities (or bodies) in a streaming fashion, i.e. being
 able to access the incoming bytes while they come in from the network without having to buffer a potentially big request
 or response in memory. The same is valid for sending out request or response data entities. In the model, the streaming
-underlyings can be seen in the @unidoc[HttpEntity] type which now has subclasses that allow to specify a @unidoc[Source[ByteString, Any]`
+underlyings can be seen in the @unidoc[HttpEntity] type which now has subclasses that allow to specify a @unidoc[Source[ByteString, Any]]
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/migration-guide/migration-from-spray.md]: `@unidoc[Source[ByteString, NotUsed]]` for consistency.

File: docs/src/main/paradox/scala/http/routing-dsl/source-streaming-support.md [hunk 5]
```diff
@@ -1,9 +1,9 @@
 # Source Streaming
 
-Akka HTTP supports completing a request with an Akka @scala[@unidoc[Source[T, _]`]@java[@unidoc[Source[T, ?]]], which makes it possible to easily build
-and consume streaming end-to-end APIs which apply back pressure throughout the entire stack. 
+Akka HTTP supports completing a request with an Akka @unidoc[Source[T, \_]], which makes it possible to easily build
+and consume streaming end-to-end APIs which apply back pressure throughout the entire stack.
 
-It is possible to complete requests with raw @scala[`Source[ByteString, _]`]@java[@unidoc[Source[ByteString, ?]]], however often it is more convenient to 
+It is possible to complete requests with raw @unidoc[Source[ByteString, \_]], however often it is more convenient to
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/routing-dsl/source-streaming-support.md]: `@unidoc[Source[ByteString, NotUsed]]` for consistency.

File: docs/src/main/paradox/scala/http/server-side/low-level-api.md [hunk 6]
```diff
@@ -25,11 +25,11 @@ implementation as well as on all levels of its API.
...
-source and, for each of them, provides a @scala[@unidoc[Flow[HttpRequest, HttpResponse, _]`]@java[@unidoc[Flow[HttpRequest, HttpResponse, ?]]] to "translate" requests into responses.
+source and, for each of them, provides a @unidoc[Flow[HttpRequest, HttpResponse, \_]] to "translate" requests into responses.
 
-Apart from regarding a socket bound on the server-side as a @scala[@unidoc[Source[IncomingConnection]`]@java[@unidoc[Source[IncomingConnection]]] and each connection as a
-@scala[`Source[HttpRequest]`]@java[@unidoc[Source[HttpRequest]]] with a @scala[@unidoc[Sink[HttpResponse]]]@java[@unidoc[Sink[HttpResponse]]] the stream abstraction is also present inside a single HTTP
-message: The entities of HTTP requests and responses are generally modeled as a @scala[`Source[ByteString]]]@java[@unidoc[Source[ByteString]]]. See also
+Apart from regarding a socket bound on the server-side as a @unidoc[Source[IncomingConnection]] and each connection as a
+@unidoc[Source[HttpRequest]] with a @unidoc[Sink[HttpResponse]] the stream abstraction is also present inside a single HTTP
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/server-side/low-level-api.md]: `@unidoc[Source[ByteString, NotUsed]]` for consistency.

**Comment by raboof** [comment]
@jonas @jrudolph I agree it might be good to standardize, `Source[T, _]` (`Source<T, ?>` for Java) seems a bit shorter/less noisy, WDYT?

**Comment by jonas** [comment]
Sounds good to me. `NotUsed` requires more knowledge of Akka than `_`/`?`.

**Comment by jrudolph** [comment]
Hmm, maybe let's use underscore / question mark, indeed. In most cases, a placeholder is actually what we want to say. Sorry for changing my opinion on this back and forth...

**Comment by jrudolph** [comment]
Does this one merge right with #1649? Or should those two be merged together?

**Comment by raboof** [comment]
This PR includes #1649, so we can either merge that one first or merge this one (and just close that one)

**Comment by raboof** [comment]
#1667

**Comment by raboof** [comment]
PLS BUILD

**Pull request merged**